homeassistant/components/rachio/switch.py

```diff
@@ -42,32 +41,33 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Rachio switches."""
     # Add all zones from all controllers as switches
-    devices = await hass.async_add_executor_job(_create_devices, hass, config_entry)
-    async_add_entities(devices)
-    _LOGGER.info("%d Rachio switch(es) added", len(devices))
+    entities = await hass.async_add_executor_job(_create_entities, hass, config_entry)
+    async_add_entities(entities)
+    _LOGGER.info("%d Rachio switch(es) added", len(entities))
 
 
-def _create_devices(hass, config_entry):
-    devices = []
+def _create_entities(hass, config_entry):
+    entities = []
     person = hass.data[DOMAIN_RACHIO][config_entry.entry_id]
     # Fetch the schedule once at startup
     # in order to avoid every zone doing it
     for controller in person.controllers:
-        devices.append(RachioStandbySwitch(hass, controller))
+        entities.append(RachioStandbySwitch(controller))
         zones = controller.list_zones()
         current_schedule = controller.current_schedule
         _LOGGER.debug("Rachio setting up zones: %s", zones)
         for zone in zones:
             _LOGGER.debug("Rachio setting up zone: %s", zone)
-            devices.append(RachioZone(hass, person, controller, zone, current_schedule))
-    return devices
+            entities.append(RachioZone(person, controller, zone, current_schedule))
+    return entities
 
 
-class RachioSwitch(SwitchDevice):
+class RachioSwitch(RachioDeviceInfoProvider, SwitchDevice):
     """Represent a Rachio state that can be toggled."""
 
     def __init__(self, controller, poll=True):
         """Initialize a new Rachio switch."""
+        super().__init__(controller)
         self._controller = controller
```

Comment:
And here.

Reply:
done

Reply:
@MartinHjelmare 
```
************* Module homeassistant.components.rachio.binary_sensor
homeassistant/components/rachio/binary_sensor.py:38:4: W0231: __init__ method from base class 'RachioDeviceInfoProvider' is not called (super-init-not-called)
************* Module homeassistant.components.rachio.switch
homeassistant/components/rachio/switch.py:68:4: W0231: __init__ method from base class 'RachioDeviceInfoProvider' is not called (super-init-not-called)

```
Sadly that made pylint unhappy.

Reply:
We should still call super, but since super sets `self._controller` we don't need to do that here.
```suggestion
```